test/jasmine/tests/plots_test.js

```diff
@@ -408,6 +408,19 @@ describe('Test Plots', function() {
         beforeEach(function(done) {
             gd = createGraphDiv();
             Plotly.plot(gd, [{ x: [1, 2, 3], y: [2, 3, 4] }], {}).then(done);
+
+            // hacky: simulate getting stuck with these flags due to an error
+            // see #2055 and commit 6a44a9a - before fixing that error, we would
+            // end up in an inconsistent state that prevented future Plotly.newPlot
+            // because _dragging and _dragged were not cleared by purge.
```

Comment:
That's the first time I'd seen an error that blocked all future `Plotly.newPlot` calls! We really need to encapsulate all these `gd.***` vars better...

Reply:
Yep, very nice. Thanks!